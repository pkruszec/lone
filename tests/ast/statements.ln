;

if 1 {
    foo();
} else {
    bar();
}

if cast(int) x == 5
    foo();
else if 0
    bar();
else
    baz();

if 0 foo(); else bar();
