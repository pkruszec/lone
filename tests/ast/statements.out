PROG
  NOP
  IF
    IF_COND
      NUM(1)
    IF_IF
      BLOCK
        EVAL
          PROC_CALL
            IDENT(foo)
    IF_ELSE
      BLOCK
        EVAL
          PROC_CALL
            IDENT(bar)
  IF
    IF_COND
      EQUAL
        CAST
          IDENT(int)
          IDENT(x)
        NUM(5)
    IF_IF
      EVAL
        PROC_CALL
          IDENT(foo)
    IF_ELSE
      IF
        IF_COND
          NUM(0)
        IF_IF
          EVAL
            PROC_CALL
              IDENT(bar)
        IF_ELSE
          EVAL
            PROC_CALL
              IDENT(baz)
  IF
    IF_COND
      NUM(0)
    IF_IF
      EVAL
        PROC_CALL
          IDENT(foo)
    IF_ELSE
      EVAL
        PROC_CALL
          IDENT(bar)
